File: Easy/2965. Find Missing and Repeated Values/Elixir/main.ex
defmodule Solution do
  @spec find_missing_and_repeated_values(grid :: [[integer]]) :: [integer]
  def find_missing_and_repeated_values(grid) do
    n = length(grid)
    range = 1..(n * n) |> Enum.to_list()
    flattened = List.flatten(grid)

    # Find the repeated number
    repeated = flattened
      |> Enum.frequencies()
      |> Enum.find(fn {_, v} -> v > 1 end)
      |> elem(0)

    # Find the missing number using MapSet
    missing = MapSet.difference(MapSet.new(range), MapSet.new(flattened))
      |> Enum.at(0)

    [repeated, missing]
  end
end

File: Easy/2965. Find Missing and Repeated Values/Python/main.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        n = len(grid);

        nums = [0] * (n * n + 1);

        for i in range(n):
            for j in range(n):
                nums[grid[i][j]] += 1;
        
        missing = -1;
        repeated = -1;

        for i in range(1, n * n + 1):
            if nums[i] == 0:
                missing = i;
            if nums[i] == 2:
                repeated = i;
        
        return [repeated, missing];

